clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecords.java

```diff
@@ -93,25 +100,47 @@ public int count() {
     private static class ConcatenatedIterable<K, V> implements Iterable<ConsumerRecord<K, V>> {
 
         private final Iterable<? extends Iterable<ConsumerRecord<K, V>>> iterables;
+        private Predicate<ConsumerRecord<K, V>> predicate = null;
 
         public ConcatenatedIterable(Iterable<? extends Iterable<ConsumerRecord<K, V>>> iterables) {
             this.iterables = iterables;
         }
 
+        public ConcatenatedIterable(Iterable<? extends Iterable<ConsumerRecord<K, V>>> iterables, Predicate<ConsumerRecord<K, V>> predicate) {
```

Comment:
That could be a issue when iterating results as you filter "all" records. It means users need to "see" all records ... Please add the scenario to benchmark.